MapboxCoreNavigation/CLLocation.swift

```diff
@@ -88,25 +88,30 @@ extension CLLocation {
         }
         
         guard let closest = Polyline(nearByCoordinates).closestCoordinate(to: coordinate) else { return nil }
-        guard let calculatedCourseForLocationOnStep = interpolatedCourse(along: nearByCoordinates) else { return nil }
+        guard let projectedLocation = Polyline(nearByCoordinates).trimmed(from: closest.coordinate, distance: projectedDistance).coordinates.last else { return nil }
```

Comment:
`interpolatedCourse(along:)` already applies a weird half-second or 15-meter dead-reckoning interval (whichever is greater). What does this additional “projected distance” do to that dead reckoning?

Reply:
This is only used for calculating the points ahead/behind the user to create an angle used for the snapped course.

Reply:
Maybe 1/2 second isn't enough. Or maybe we were also suffering from delays due to unrelated performance issues. I'm really not sure, which is why I want to try to measure this in the field today.